profile/src/v100/address/resource_address.rs

```diff
@@ -0,0 +1,133 @@
+use crate::prelude::*;
+
+/// The address of an Account, a bech32 encoding of a public key hash
+/// that starts with the prefix `"account_"`, dependent on NetworkID, meaning the same
+/// public key used for two AccountAddresses on two different networks will not have
+/// the same address.
+#[derive(
+    Clone,
+    Debug,
+    Default,
+    PartialEq,
+    Eq,
+    Hash,
+    PartialOrd,
+    Ord,
+    SerializeDisplay,
+    DeserializeFromStr,
+    derive_more::Display,
+    uniffi::Record,
+)]
+#[display("{address}")]
+pub struct ResourceAddress {
+    pub address: String,
+    pub network_id: NetworkID,
+}
+
+#[uniffi::export]
+pub fn new_resource_address(bech32: String) -> Result<ResourceAddress> {
```

Comment:
I've chosen the naming convention to start all global freestanding functions which are actually "constructors" with `new_`